### DISCUSSION TITLE
how to fix the garbled code

### DISCUSSION THREAD

**Discussion by ytcpub** [pymupdf/PyMuPDF#4016] [Q&A]
[garbled_26282171_1.pdf](https://github.com/user-attachments/files/17610659/garbled_26282171_1.pdf)

in above pdf, I get some garbled '5\x17\x04\x07\x08;\x02?\x12\x12\x02@9A\x13\x0231\x0231\x02\x111\x02!\x12\x02+\x06\x05\x02;\x02?\x12\x12\x02@9A\x13\x02!\x15\x02\x15!\x02!1\x02\x11!\x02\n\x04\x04\x03',  and some normal eg 'on September 16, 2015. © 2015 Am..'
how to get the right content? it seems the encoding problem , but why some is right, and some is \x...

``` python
doc = fitz.open('garbled_26282171_1.pdf')
page = doc[0]
txt_blocks = page.get_text("dict", flags=fitz.TEXTFLAGS_TEXT, sort=True)['blocks']

```

**Comment by JorjMcKie** [comment]
This is a problem caused by a font. Text extraction can only work if a complete and correct back-translation table from the "glyph" (program that writes the visible character) to the originating Unicode exists.
This needs not to be the case at all!
It therefore is **_not an encoding problem!_**
Just missing or incomplete backtranslation information.
Your only way out is OCR-ing that page or the respective text bboxes. To know which they are, make sure to switch off the text extraction flag `pymupdf.TEXT_CID_FOR_UNKNOWN_UNICODE` (part of all default extraction flags). Your extracted text will then contain &#xfffd; characters for those misses - instead of results of MuPDF's replacement efforts. Pass the respective bbox to OCR and continue with the result. An example script for this approach can be found [here](https://github.com/pymupdf/PyMuPDF-Utilities/blob/master/OCR/tesseract2.py).

**Reply by ytcpub** [reply]
got, thanks very much
